### DISCUSSION TITLE
Older image when download.

### DISCUSSION THREAD

**Discussion by ns-bcr** [open-webui/open-webui#7248] [Issues]
# Bug Report

---

## Installation Method

I installed Open WebUI using Docker with the following command:  
```bash
docker run -d -p 3000:8080 --gpus all -v open-webui:/app/backend/data --name open-webui ghcr.io/open-webui/open-webui:cuda
```

---

## Environment

- **Open WebUI Version:** v0.3.32  
- **Operating System:** Ubuntu 24.4.0  
- **Browser (if applicable):** Opera GX  

---

## Confirmation

- [x] I have read and followed all the instructions provided in `README.md`.  
- [ ] I am using the latest versions of both Open WebUI and Ollama.  
- [x] I have attached browser console logs.  
- [ ] I have attached the Docker container logs.  
- [x] I have included detailed reproduction steps in the "Reproduction Details" section.  

---

## Expected Behavior

I noticed a new version (v0.4.3) was available. Initially, I couldn’t find the update page. To resolve this, I removed my existing Docker instance and reinstalled the Docker image.

---

## Actual Behavior

Instead of getting the latest version (v0.4.3), I ended up with an older version than what I had previously installed (v0.3.32).  

Additionally:  
- I lost all my data. Thankfully, I had backed up most of it.  
- However, I am unable to download the `config.yaml` file.  

---

## Bug Summary

This issue might be related to a missing migration step. I saw a reference to migrations in the documentation, but since I can’t access the `config.yaml`, I couldn’t proceed.

---

### Steps to Reproduce:

1. Pull the latest Docker image:  
   ```bash
   docker pull ghcr.io/open-webui/open-webui:main
   ```
2. Run the container:  
   ```bash
   docker run -d -p 3000:8080 --gpus all -v open-webui:/app/backend/data --name open-webui ghcr.io/open-webui/open-webui:cuda
   ```

---

## Logs and Screenshots

**Browser Console Logs:**  
![Browser Console Logs Screenshot](https://github.com/user-attachments/assets/98303975-6a7a-4294-a541-8d114e126989)  

---

## Additional Note

It seems the issue might be resolved through a migration step, but without access to the `config.yaml` file, I am unable to confirm this.

**Comment by alkaris2** [comment]
I get this exact issue also, except it returns `Error 404`

If trying to visit the url `https://ghcr.io/v2/open-webui/manifests/main` so see if it actually responds, it just returns;

```
code	"UNAUTHORIZED"
message	"authentication required"
```

the URL for `https://ghcr.io` is defunct and not accessible when trying to download the update. GitHub seems to be forcing logins when using their short URL handler. So a different update URL needs to be provided, probably direct download link from the Releases page.

also running update from Docker returns the following;

```BASH
$ docker pull ghcr.io/open-webui:main
Error response from daemon: Head "https://ghcr.io/v2/open-webui/manifests/main": name invalid

$ docker pull ghcr.io/open-webui:cuda
Error response from daemon: Head "https://ghcr.io/v2/open-webui/manifests/main": name invalid
```

**Comment by mahenning** [comment]
This seems to be fixed, at least [open-webui:cuda](https://github.com/open-webui/open-webui/pkgs/container/open-webui/318633120?tag=cuda) points to a build 4 days ago. When downloading the image and looking into the CHANGELOG.md file in it, the latest entry is for version 0.4.8, and it is also the version on starting the app. Maybe at the time of your issue the tagging was behind on the "cuda" tag? Should still not happen.
Also maybe a stupid question but did you remove your current `open-webui:cuda` **image** before re-creating the container? If you only removed the container and used the docker run command above to re-create it, docker would still use the local (maybe old) image.

@alkaris2 It seems you tried to pull from a wrong URL. Instead of `ghcr.io/open-webui:main` try using `ghcr.io/open-webui/open-webui:main`.
